index.js

```diff
@@ -186,14 +186,27 @@ class PSPDFKitView extends React.Component {
    * Adds a new annotation to the current document.
    *
    * @param annotation InstantJson of the annotation to add.
+   * 
+   * Returns a promise resolving a dictionary with the following structure:
+   * {'annotation' : instantJson}.
```

Comment:
On iOS, the promise resolves with a boolean (true) when the annotation has been successfully added or rejects with `{"error", "Failed to add annotation."}`.

Reply:
BTW, we have the same behavior for all annotation manipulations methods.

Reply:
I'm gonna change the success result to be just a boolean but I think it makes sense to return the actual error that was thrown instead of a generic one.

Reply:
> I'm gonna change the success result to be just a boolean but I think it makes sense to return the actual error that was thrown instead of a generic one.

That is a very good point! I created a followup issue for that https://github.com/PSPDFKit/react-native/issues/260